### PR TITLE
Feature/99183 internal case status

### PR DESCRIPTION
**Pull request by marijnritense**

### CODE REVIEW

File: document/src/main/kotlin/com/ritense/document/domain/InternalCaseStatusId.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+package com.ritense.document.domain
+
+import com.ritense.valtimo.contract.domain.AbstractId
+import jakarta.persistence.Column
+import jakarta.persistence.Embeddable
+
+
+@Embeddable
+data class InternalCaseStatusId(
+    @Column(name = "case_definition_name", length = 50, columnDefinition = "VARCHAR(50)")
```
Comment (Klaas-Ritense) [inline document/src/main/kotlin/com/ritense/document/domain/InternalCaseStatusId.kt]: The `case_definition_name` can be 256 long

File: document/src/main/kotlin/com/ritense/document/service/InternalCaseStatusService.kt [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import kotlin.jvm.optionals.getOrNull
+
+@Transactional
+class InternalCaseStatusService(
+    private val internalCaseStatusRepository: InternalCaseStatusRepository,
+    private val documentDefinitionService: DocumentDefinitionService,
+    private val authorizationService: AuthorizationService,
+) {
+    fun getInternalCaseStatuses(documentDefinitionName: String): List<InternalCaseStatus> {
+        denyManagementOperation()
```
Comment (Klaas-Ritense) [inline document/src/main/kotlin/com/ritense/document/service/InternalCaseStatusService.kt:40]: Is a DENY here the right way to go, when getting the status?

Comment (marijnritense) [reply document/src/main/kotlin/com/ritense/document/service/InternalCaseStatusService.kt:40]: Yes, this is an admin-only method. Another story revolves around the case-list, so it's possible this needs to be changed then.

File: document/src/main/kotlin/com/ritense/document/domain/InternalCaseStatusId.kt (hunk 1 above)
Comment (marijnritense) [reply document/src/main/kotlin/com/ritense/document/domain/InternalCaseStatusId.kt]: fixed

Comment (ivo-ritense) [reply document/src/main/kotlin/com/ritense/document/domain/InternalCaseStatusId.kt]: As far as i can find it is only 50

Comment (Klaas-Ritense) [reply document/src/main/kotlin/com/ritense/document/domain/InternalCaseStatusId.kt]: I was wrong. It should be 50 after all

Comment (marijnritense) [reply document/src/main/kotlin/com/ritense/document/domain/InternalCaseStatusId.kt]: fixed

**Review by Klaas-Ritense — APPROVED** [review]

**Pull request merged**